operations/dnsmasq/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM alpine:latest
+
+# Install dnsmasq
+RUN apk --no-cache add dnsmasq
+
+# Expose the DNS port
+EXPOSE 53/udp
+
+# Place the environment's config in the config directory
+ARG AZ_ENV
+COPY config /tmp/config
+RUN cp -r /tmp/config/${AZ_ENV}/. /etc/dnsmasq.d/
```

Comment:
should we delete `/tmp/config` after doing this copy? just to make sure configurations from other environments aren't on here (even in a weird place)

Reply:
I like it! Added!

FYI, this method of baking the configuration into the container, I dislike. Doing so I am choosing implementation speed over maintainability.

I think the optimal solution would involve an environment-agnostic Docker container with the configuration provided by a bind mount. Just doing so involves a few implementation details that need be thought through, that I didn't want to tackle at this time.

Namely, storage accounts (which would be used for the bind mount) are created after our core network infrastructure. Accessing the storage accounts needs the VPN, which then needs this DNS server. I think to get to bind mounts we'd need to create a separate Terraform module.

Reply:
True, but this container is used only for our VPN connection. DNS resolution on Azure is mess. This document explains how it works in various situations: https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-name-resolution-for-vms-and-role-instances

TLDR;

The Azure DNS server `168.63.129.16` will return our private DNS records when resolved from within the VNET. However, when resolved outside the VNET, for example, through the VPN Gateway, the Azure DNS server is unable to identify our DNS queries as from the VNET, so the public query records are used.

To resolve that, Microsoft recommends setting up a DNS forwarder in your VNET, which we are doing via dnsmasq. By setting up a forwarded, we have a VNET IP we can point the VPN clients at and Azure's VPN server sees the requests coming from our VNET and returns the private IP responses.

Prior to this PR, we were good with just forwarding functionality, but now we need to have some control over the DNS results due to multiple VNETs and only one of them being accessible via the VPN (we can't access the CDC network via our VPN).